### PR TITLE
Split usage samples in README into sections.

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Helps with readability and also allows users to make a link to particular section!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,40 +18,61 @@ The goal of this implementation is to be convenient to use and fit Kotlin's null
 
 ### Usage
 
-```kotlin
+#### Create Some
```
Comment (ming13) [inline README.md]: `Some`?

File: README.md [hunk 2]
```diff
@@ -18,40 +18,61 @@ The goal of this implementation is to be convenient to use and fit Kotlin's null
...
 
-// Create Some.
+```kotlin
 val some = Some(value)
+```
+#### Use None
```
Comment (ming13) [inline README.md]: `None`?

File: README.md [hunk 3]
```diff
@@ -18,40 +18,61 @@ The goal of this implementation is to be convenient to use and fit Kotlin's null
...
+
+```kotlin
 val values: Observable<String> = Observable
     .just(Some("a"), None, Some("b"))
     .filterSome()
+    .map { value: String -> } // filterSome() unwraps value for you.
+```
 
-// Filter only None values emitted by RxJava2 Observable.
+#### Filter only `None` values emitted by RxJava2/ProjectReactor
```
Comment (ming13) [inline README.md]: RxJava 2 or Project Reactor

**Comment by ming13** [comment]
Maybe fix the remaining styling issues? 

* Get value or fall back to something else if `None` (`getOrElse()`)
* Filter only `None` values emitted by RxJava 2 or Project Reactor
* Check if `Optional` is `Some` or `None`

File: README.md [hunk 4]
```diff
@@ -18,40 +18,61 @@ The goal of this implementation is to be convenient to use and fit Kotlin's null
...
+#### Convert `Optional<T>` to `T?`
 
-// Convert Optional<T> to T?.
+```kotlin
 // If Optional is None — you'll get null, otherwise you'll get not null T value.
 val t = optional.toNullable()
+```
 
-// Get value or fallback to other if None (getOrElse()).
+#### Get value or fall back to something else if `None` (getOrElse())
```
Comment (nostra13) [inline README.md]: `fallback`?

Comment (artem-zinnatullin) [reply README.md]: Consider `fall back` suk

Comment (nostra13) [reply README.md]: Consider `osenь spina` pidr

Comment (artem-zinnatullin) [reply README.md]: tobi 3.14sta

**Pull request merged**